rmw/src/discovery_options.c

```diff
@@ -21,10 +21,10 @@
 rmw_discovery_options_t
 rmw_get_zero_initialized_discovery_options(void)
 {
-  rmw_discovery_options_t result = (rmw_discovery_options_t) {
+  static const rmw_discovery_options_t result = {
     .automatic_discovery_range = RMW_AUTOMATIC_DISCOVERY_RANGE_NOT_SET,
     .static_peers_count = 0,
```

Comment:
So actually, RMW_AUTOMATIC_DISCOVERY_RANGE_NOT_SET is 0, so we can use the universal specifier here if we want:

```suggestion
  // All members are initialized to 0 or NULL by C99 6.7.8/10.
  static const rmw_discovery_options_t result;
```

Reply:
i was thinking that once `RMW_AUTOMATIC_DISCOVERY_RANGE_NOT_SET` enum is updated to another value in the future, this breaks easily. (that i think we should do that but guaranteed.) So that was my intention to leave this here. (i guess original author though about this in the same way)

Reply:
While that is true, that shouldn't matter here.  This is the `get_zero_initialized` function, so it should always return 0 (in my opinion).

Reply:
yeah `RMW_AUTOMATIC_DISCOVERY_RANGE_NOT_SET ` should have been used with `xxx_get_default_option`. that is what i will do with the another patch, i am okay with 0 init then.